lib/matplotlib/axes/_axes.py

```diff
@@ -4942,6 +4946,16 @@ def reduce_C_function(C: array) -> float
         tx = x
         ty = y
 
+        xscale_kwargs = {'base': 10, 'linthresh': 2, 'linscale': 1}
+        if isinstance(xscale, tuple):
+            xscale, xscale_kwargs_ = xscale
+            xscale_kwargs = xscale_kwargs_.copy()
```

Comment:
This doesn't do what @timhoffm suggested; it replaces `xscale_kwargs` and then loses any defaults that might have been there.
```suggestion
        xscale_kwargs = {'base': 10, 'linthresh': 2, 'linscale': 1}
        if isinstance(xscale, tuple):
            xscale, xscale_kwargs_ = xscale
            xscale_kwargs.update(xscale_kwargs_)
```